packages/react/src/patterns/sub-patterns/TableOfContents/TOCMobile.js

```diff
@@ -92,14 +93,8 @@ const TOCMobile = ({ menuItems, selectedId, menuLabel, updateState }) => {
     });
     const title = filteredItems[0].title;
     updateState(id, title);
-    document.querySelector(`a[name="${id}"]`).scrollIntoView(true);
-    const scrolledY = root.scrollY;
-    if (scrolledY) {
-      root.scroll({
-        top: scrolledY - 48,
-        behavior: 'smooth',
-      });
-    }
+    const selector = `a[name="${id}"]`;
+    smoothScroll({ e: null, selector });
```

Comment:
Based on the logic below I don't think `e: null` is necessary. Is that right?

Reply:
@photodow If we did not pass an event with null argument in the object it is taking event as not null so and then looking for `href`. And I believe `e: null` is necessary that is how I am handling in utility.

Reply:
I see then what if we removed the `null` requirement in the utility and just check if it was passed or not `!e`?

Reply:
@photodow updated the PR